### PR TITLE
Add function to calculate soap descriptors

### PR DESCRIPTION
**Pull request by jan-janssen**

### CODE REVIEW

**Review by pmrv — APPROVED** [review]
Been meaning to add DScribe myself, so I like this.  docstring wouldn't hurt though.

**Review by liamhuber — COMMENTED** [review]
Docstring is also critical for a new tool, but for a tool like this it could be as simple as "Creates a `dscribe.SOAP` object and creates descriptors from it, as documented below (if `dscribe` is available:" and then (conditionally on `dscribe` being available) append the the docs from `SOAP` and `SOAP.create` to the function's `__doc__` attribute. Actually, I guess the `structure` arg should still be documented, but the rest line up perfectly with the wrapped library so there's no need to re-write things here.

File: structuretoolkit/analyse/dscribe.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        sparse=sparse,
+        dtype=dtype,
+    )
+    return periodic_soap.create(
+        system=structure,
+        centers=centers,
+        n_jobs=n_jobs,
+        only_physical_cores=only_physical_cores,
+        verbose=verbose,
+    )
```
Comment (ligerzero-ai) [inline structuretoolkit/analyse/dscribe.py:44]: I prefer 

```
structure_soap_array = periodic_soap.create(...)

return structure_soap_array 
```

More explicit and clearer to someone who is trying to figure out what the function does.

Comment (liamhuber) [reply structuretoolkit/analyse/dscribe.py:44]: That would actually integrate better with the workflow stuff too. There, we scrape the output channel label from whatever's after the return value -- giving it a nice variable like that keeps things extra tidy.

**Comment by jan-janssen** [comment]
Check if it is possible to build `dscribe`  for Windows https://github.com/conda-forge/dscribe-feedstock/pull/36

**Pull request merged**